File: backend-app/src/main/java/tom/projects/monkeylog/controller/MeasurementController.java
package tom.projects.monkeylog.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import tom.projects.monkeylog.dto.measurement.*;
import tom.projects.monkeylog.mapper.MeasurementMapper;
import tom.projects.monkeylog.service.MeasurementService;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class MeasurementController {
    private final MeasurementService measurementService;
    private final MeasurementMapper measurementMapper;

    @GetMapping("/measurement")
    List<MeasurementFullResponse> all() {
        return measurementMapper.measurementsToMeasurementFullResponses(measurementService.all());
    }

    @PostMapping("/measurement")
    MeasurementResponse create(@RequestBody MeasurementCreateRequest measurementCreateRequest) {
        return measurementMapper.measurementToMeasurementResponse(measurementService.create(measurementCreateRequest));
    }

    @PatchMapping("/measurement/id")
    MeasurementResponse update(@RequestBody MeasurementUpdateRequest measurementUpdateRequest, @PathVariable Long id) {
        return measurementMapper.measurementToMeasurementResponse(measurementService.update(measurementUpdateRequest, id));
    }

    @DeleteMapping("/measurement/{id}")
    void delete(@PathVariable Long id) {
        measurementService.delete(id);
    }

    @PostMapping("/measurement/{id}/measurement_point")
    MeasurementPointResponse createPoint(@RequestBody MeasurementPointCreateRequest measurementPointCreateRequest, @PathVariable Long id) {
        return measurementMapper.measurementPointToMeasurementPointResponse(measurementService.createPoint(id, measurementPointCreateRequest));
    }

    @PatchMapping("/measurement_point/{id}")
    MeasurementPointResponse updateField(@RequestBody MeasurementPointUpdateRequest measurementPointUpdateRequest, @PathVariable Long id) {
        return measurementMapper.measurementPointToMeasurementPointResponse(measurementService.updatePoint(id, measurementPointUpdateRequest));
    }

    @DeleteMapping("/measurement_point/{id}")
    void destroyPoint(@PathVariable Long id) {
        measurementService.deletePoint(id);
    }
}

File: backend/src/main/kotlin/tom/com/workout/logbook/controller/MeasurementController.kt
package tom.com.workout.logbook.controller

import io.swagger.v3.oas.annotations.tags.Tag
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import tom.com.workout.logbook.dto.measurement.*
import tom.com.workout.logbook.mapper.toFullResponse
import tom.com.workout.logbook.mapper.toResponse
import tom.com.workout.logbook.model.measurement.Measurement
import tom.com.workout.logbook.service.MeasurementService
import java.util.*

@RestController
@Tag(name = "Measurement")
class MeasurementController(private val measurementService: MeasurementService) {
    @GetMapping("/measurements")
    fun getMeasurements() = measurementService.all().map(Measurement::toFullResponse)

    @GetMapping("/measurements/{id}")
    fun getMeasurement(@PathVariable id: UUID) = measurementService.get(id).toFullResponse()

    @PostMapping("/measurements")
    fun createMeasurement(@RequestBody measurementCreateRequest: @Valid MeasurementCreateRequest) =
        measurementService.create(measurementCreateRequest).toResponse()

    @PatchMapping("/measurements/{id}")
    fun updateMeasurement(
        @RequestBody measurementUpdateRequest: @Valid MeasurementUpdateRequest,
        @PathVariable id: UUID
    ) = measurementService.update(measurementUpdateRequest, id).toResponse()

    @DeleteMapping("/measurements/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteMeasurement(@PathVariable id: UUID) = measurementService.delete(id)

    @PostMapping("/measurements/{id}/measurement-points")
    fun createMeasurementPoint(
        @RequestBody @Valid measurementPointCreateRequest: MeasurementPointCreateRequest,
        @PathVariable id: UUID
    ) = measurementService.createPoint(id, measurementPointCreateRequest).toResponse()

    @PatchMapping("/measurement-points/{id}")
    fun updateMeasurementPoint(
        @RequestBody @Valid measurementPointUpdateRequest: MeasurementPointUpdateRequest,
        @PathVariable id: UUID
    ) = measurementService.updatePoint(id, measurementPointUpdateRequest).toResponse()


    @DeleteMapping("/measurement-points/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteMeasurementPoint(@PathVariable id: UUID) = measurementService.deletePoint(id)
}